mm2src/lp_swap/maker_swap.rs

```diff
@@ -691,9 +711,44 @@ impl MakerSwap {
             tx_hash,
         };
 
-        Ok((Some(MakerSwapCommand::Finish), vec![MakerSwapEvent::TakerPaymentSpent(
-            tx_ident,
-        )]))
+        Ok((Some(MakerSwapCommand::ConfirmTakerPaymentSpend), vec![
+            MakerSwapEvent::TakerPaymentSpent(tx_ident),
+            MakerSwapEvent::TakerPaymentSpendConfirmStarted,
+        ]))
+    }
+
+    async fn confirm_taker_payment_spend(&self) -> Result<(Option<MakerSwapCommand>, Vec<MakerSwapEvent>), String> {
+        // check if the taker coin requires confirmations
+        if self.r().data.taker_payment_confirmations < 1 {
```

Comment:
I think we should still wait for 1 confirmation of spend even if `taker_payment_confirmations < 1`. For example we can have 0 confirmations for ETH, if the payment spend fails for some reason in this case we won't detect it and treat the swap as successfully finished which would be wrong.

Reply:
Thanks, removed this checking :)